### PR TITLE
chore: comments, logs, & cleanup

### LINKED ISSUE #262: Error and Doc audit: first pass
# Summary

We need to do an entire error and doc audit of the codebase, particularly the `homestar-runtime`. 

This includes:

- [x] correcting levels and which [tracing](https://docs.rs/tracing/latest/tracing/index.html#using-the-macros) field names we want as top-level "keys."
- [x] Adding category and subject fields to all logs. 
- [x] Even though the runtime should leverage `anyhow`, and does, we should designate specific error types to bubble up things throughout the runtime and capture fields we care about.
~~- [ ] Use [`anyhow::context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) more judiciously for error handling contextual information.~~ -> moved to https://github.com/ipvm-wg/homestar/issues/455
~~- [ ] Handle backtraces in logs and other formatting concerns for logfmt.~~ -> moved to https://github.com/ipvm-wg/homestar/issues/455)
- [X] Correct `expect` phrases.
- [X] allow_dead_code checks/rechecks
- [X] pub/private/visibility checks

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
# Description

Remove many TODOs, add comments on *pub* (*pub* crate), and work through logs.

- Includes a few other minor fixins and additions. 

## Type of change

- [X] Refactor (non-breaking change that updates existing functionality)

Closes #262.

### CODE REVIEW

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 1]
```diff
@@ -222,7 +245,11 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
 
                         // Skip dialing peers if at connected peers limit
                         if connected_peers_count >= event_handler.connections.max_peers as usize {
-                            warn!("peers discovered through rendezvous not dialed because max connected peers limit reached");
+                            debug!(
```
Comment (bgins) [inline homestar-runtime/src/event_handler/swarm_event.rs:248]:
```suggestion
                            warn!(
```

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 2]
```diff
@@ -263,14 +290,18 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
...
+                                              err=?err,
+                                              "failed to dial discovered peer");
                                     }
                                 };
                             } else if !self_registration {
-                                warn!(
-                                        peer_id=registration.record.peer_id().to_string(),
-                                        "peer discovered through rendezvous not dialed because the max connected peers limit was reached"
-                                    )
+                                debug!(subject = "libp2p.rendezvous.cli.discovered",
```
Comment (bgins) [inline homestar-runtime/src/event_handler/swarm_event.rs:301]:
```suggestion
                                warn!(subject = "libp2p.rendezvous.cli.discovered",
```

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 3]
```diff
@@ -439,16 +499,19 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
                             },
                         );
                     }
-                    Err(err) => info!(err=?err, "cannot handle incoming gossipsub message"),
+                    Err(err) => debug!(subject = "libp2p.gossipsub.err",
```
Comment (bgins) [inline homestar-runtime/src/event_handler/swarm_event.rs:502]:
```suggestion
                    Err(err) => warn!(subject = "libp2p.gossipsub.err",
```

File: homestar-runtime/src/network/swarm.rs [hunk 4]
```diff
@@ -42,7 +43,12 @@ pub(crate) async fn new(settings: &settings::Network) -> Result<Swarm<ComposedBe
         .with_context(|| "failed to generate/import keypair for libp2p".to_string())?;
 
     let peer_id = keypair.public().to_peer_id();
-    info!(peer_id = peer_id.to_string(), "local peer ID generated");
+    info!(
+        subject = "swarm.init.err",
```
Comment (bgins) [inline homestar-runtime/src/network/swarm.rs:47]:
```suggestion
        subject = "swarm.init",
```

File: homestar-runtime/src/network/webserver.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-//! Sets up a websocket server for sending and receiving messages from browser
+//! Sets up a webserver for websocket and HTTP interaction with external
```
Comment (bgins) [inline homestar-runtime/src/network/webserver.rs:1]:
```suggestion
//! Sets up a webserver for websocket and HTTP interaction with
```

Possible edit. All clients are external.

File: homestar-runtime/src/network/webserver.rs [hunk 6]
```diff
@@ -111,6 +121,8 @@ impl Server {
         broadcast::channel(capacity)
     }
 
+    /// Set up a new [Server] instance, which simultaneously acts as both a
```
Comment (bgins) [inline homestar-runtime/src/network/webserver.rs:124]:
```suggestion
    /// Set up a new [Server] instance, which acts as both a
```

File: homestar-runtime/src/network/webserver/rpc.rs [hunk 7]
```diff
@@ -56,7 +58,7 @@ pub(crate) const SUBSCRIBE_NETWORK_EVENTS_ENDPOINT: &str = "subscribe_network_ev
 #[cfg(feature = "websocket-notify")]
 pub(crate) const UNSUBSCRIBE_NETWORK_EVENTS_ENDPOINT: &str = "unsubscribe_network_events";
 
-/// TODO
+/// Context needed for RPC methods.
```
Comment (bgins) [inline homestar-runtime/src/network/webserver/rpc.rs:61]:
```suggestion
/// Context for RPC methods.
```

File: homestar-runtime/src/network/webserver/rpc.rs [hunk 8]
```diff
@@ -67,7 +69,7 @@ pub(crate) struct Context {
     workflow_listeners: Arc<DashMap<SubscriptionId<'static>, (Cid, FastStr)>>,
 }
 
-/// TODO
+/// Context needed for RPC methods.
```
Comment (bgins) [inline homestar-runtime/src/network/webserver/rpc.rs:72]:
```suggestion
/// Context for RPC methods.
```

File: homestar-runtime/src/worker.rs [hunk 9]
```diff
@@ -152,7 +168,10 @@ where
         {
             Ok(ctx) => self.run_queue(ctx.scheduler, running_tasks).await,
             Err(err) => {
-                error!(err=?err, "error initializing scheduler");
+                error!(subject = "worker.init",
```
Comment (bgins) [inline homestar-runtime/src/worker.rs:171]:
```suggestion
                error!(subject = "worker.init.err",
```

File: homestar-runtime/src/worker.rs [hunk 10]
```diff
@@ -350,7 +387,10 @@ where
                             let resolved = match resolved.await {
                                 Ok(inst_result) => inst_result,
                                 Err(err) => {
-                                    error!(err=?err, "error resolving cid");
+                                    error!(subject = "worker.resolve_cid",
```
Comment (bgins) [inline homestar-runtime/src/worker.rs:390]:
```suggestion
                                    error!(subject = "worker.resolve_cid.err",
```

**Review by bgins — APPROVED** [review]
Looks good! Suggested a bunch of micro-changes, but this is overall a nice cleanup! ✨

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 11]
```diff
@@ -298,22 +329,31 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
                             .await;
                     } else {
                         // Do not dial peers that are not using our namespace
-                        warn!(peer_id=rendezvous_node.to_string(), namespace=?cookie.namespace(), "rendezvous peer gave records from an unexpected namespace");
+                        debug!(subject = "libp2p.rendezvous.cli.discovered",
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply homestar-runtime/src/event_handler/swarm_event.rs:332]: This seems more like debug, as we really do nothing here.

[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply homestar-runtime/src/event_handler/swarm_event.rs:332]: ah but need to fix cli -> client. Will do.

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 12]
```diff
@@ -717,9 +824,10 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
                             .build(),
                     );
                 } else {
-                    warn!(
-                        peer_id = peer_id.to_string(),
-                        "peer discovered by mDNS not dialed because max connected peers limit reached"
+                    debug!(subject = "libp2p.mdns.discovered",
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply homestar-runtime/src/event_handler/swarm_event.rs:827]: will fix the .err

File: homestar-runtime/src/event_handler/swarm_event.rs [hunk 13]
```diff
@@ -562,42 +638,67 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
...
+                                    subject = "libp2p.kad.req_resp.err",
+                                    category = "handle_swarm_event",
+                                    "not a valid provider record tag: {capsule_tag}",
+                                )
                             }
                         }
-                        None => {
-                            info!("No provider found for outbound query {id:?}")
-                        }
+                        None => debug!(
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply homestar-runtime/src/event_handler/swarm_event.rs:648]: Mainly, nobody's there, sorry.

File: homestar-runtime/src/event_handler/event.rs [hunk 14]
```diff
@@ -183,7 +187,12 @@ impl Event {
                 }
             }
             Event::Providers(Err(err)) => {
-                error!("failed to find providers: {}", err);
+                info!(
```
[the remark this answers is not in the document]
Comment (zeeshanlakhani) [reply homestar-runtime/src/event_handler/event.rs:190]: `info` b/c it just attempts.

**Pull request merged**